google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1630,11 +1655,31 @@ public OpenTelemetry getOpenTelemetry() {
 
   @Override
   public ApiTracerFactory getApiTracerFactory() {
-    List<ApiTracerFactory> apiTracerFactories = new ArrayList();
+    return createApiTracerFactory(false, false, false);
```

Comment:
A get-method should preferably not call something that creates a new instance every time it is called. The original implementation of this did create a new list every time it was called, but it did not create any of the elements in the list. Can we make sure that this method still does not create a new instance of ApiTracerFactory each time it is called?

Reply:
Yes, even the metricstracerfactory getting added is only created once. We are using the static instance of `BuiltInOpenTelemetryMetricsProvider` to call `getOrCreateOpenTelemetry` . We have a check in this if `openTelemetry` is already created, then we do not create it again. 

It will create a new `MetricsTracerFactory` object with the same `openTelemetry` object similar to `OpenTelemetryApiTracerFactory`